actionpack/test/controller/params_wrapper_test.rb

```diff
@@ -269,6 +269,27 @@ def User.nested_attributes_options
       end
     end
   end
+
+  def test_derived_wrapped_keys_from_has_and_belongs_to_many_association
+    require "active_record/reflection"
+
+    # User has_and_belongs_to_many :roles
+    def User.reflections
+      active_record_mock = MiniTest::Mock.new
+      active_record_mock.expect(:pluralize_table_names, true)
+      {
+        roles: ActiveRecord::Reflection::HasAndBelongsToManyReflection.new(:roles, {}, {}, active_record_mock)
```

Comment:
Having to require this class is a smell that this should not be in Action Pack. Maybe we should check of something the reflection has, not which reflection it is?